dotnet/src/SemanticKernel.Abstractions/SkillDefinition/ISKFunction.cs

```diff
@@ -39,7 +40,7 @@ public interface ISKFunction
     /// <summary>
     /// AI service settings
     /// </summary>
-    CompleteRequestSettings RequestSettings { get; }
+    JsonObject ServiceSettings { get; }
```

Comment:
same comment here. There are other ways to deal with custom config settings, e.g. using generics

Reply:
@dluc Thanks for review! Could you please provide an example how we could achieve the flexibility in settings (i.e. support different set of fields for different model types) using generics? We assume that there are a lot of different model types and settings schema can be changed in the future, and we can't predict what the format would be.